src/app/teacher/edit-tag/edit-tag.component.ts

```diff
@@ -24,30 +26,35 @@ import { FlexLayoutModule } from '@angular/flex-layout';
     CommonModule,
     FormsModule,
     FlexLayoutModule,
+    MatButtonModule,
     MatFormFieldModule,
     MatInputModule,
     ReactiveFormsModule
   ]
 })
 export class EditTagComponent {
-  @Input() colorControl: FormControl;
-  @Input() nameControl: FormControl;
+  @Output() closeEvent: EventEmitter<void> = new EventEmitter();
+  @Output() enterKeyEvent: EventEmitter<void> = new EventEmitter();
+  @ViewChild('nameInput') nameInput: ElementRef;
+  protected submitLabel: string = $localize`Create`;
   @Input() tag: Tag;
   private tags: Tag[] = [];
-  @Output() enterKeyEvent: EventEmitter<void> = new EventEmitter();
 
-  constructor(private projectTagService: ProjectTagService) {}
+  protected colorControl = new FormControl('');
+  protected nameControl = new FormControl('', [
+    Validators.required,
+    this.createArchivedTagValidator(),
+    this.createUniqueTagValidator()
+  ]);
+
+  constructor(protected projectTagService: ProjectTagService, protected snackBar: MatSnackBar) {}
```

Comment:
Should these be private to the class?
```suggestion
  constructor(private projectTagService: ProjectTagService, private snackBar: MatSnackBar) {}
```